server/social.coffee

```diff
@@ -387,22 +387,40 @@ module.exports = exports = (log, loga, argv) ->
     if argv.restricted?
 
       allowedToView = (req) ->
-        allowed = []
         if argv.allowed_domains?
           if Array.isArray(argv.allowed_domains)
-            allowed = argv.allowed_domains
+            allowed_domains = argv.allowed_domains
           else
             # accommodate copy bug to be fixed soon
             # https://github.com/fedwiki/wiki/blob/4c6eee69e78c1ba3f3fc8d61f4450f70afb78f10/farm.coffee#L98-L103
             for k, v of argv.allowed_domains
-              allowed.push v
-        # emails = [ { value: 'ward.cunningham@gmail.com', type: 'account' } ]
-        emails = req.session?.passport?.user?.google?.emails
-        return false unless emails
-        for entry in emails
-          have = entry.value.split('@')[1]
-          for want in allowed
-            return true if want == have
+              allowed_domains.push v
+          try
+            # emails = [ { value: 'ward.cunningham@gmail.com', type: 'account' } ]
+            emails = req.session.passport.user.google.emails
+            for entry in emails
+              have = entry.value.split('@')[1]
+              for want in allowed_domains
+                return true if want == have
+          catch error
+            return false
+        if argv.allowed_usernames?
```

Comment:
On my more recent commits, I changed the code so that both `allowed_domains` and `allowed_usernames` can exist at the same time and neither will be ignored...